### PR TITLE
[CodingStyle] Escape percent when encapsed strings to sprintf

### PR DESCRIPTION
**Pull request by zingimmick**
Fixes rectorphp/rector#6520

### CODE REVIEW

**Comment by zingimmick** [comment]
This test fixture already added at `0.9.8`
https://github.com/rectorphp/rector-src/blob/58d97b985373b480ecdcde5fa8e200dcc1f01fbc/rules/coding-style/tests/Rector/Encapsed/EncapsedStringsToSprintfRector/Fixture/encapsed_strings_to_sprintf_should_escape_percent.php.inc#L22
but missing escape percent `0.10.0` again
https://github.com/rectorphp/rector-src/blob/7e9a8f79d1c988bbc9422917bd827b40b751b847/rules-tests/CodingStyle/Rector/Encapsed/EncapsedStringsToSprintfRector/Fixture/encapsed_strings_to_sprintf_should_escape_percent.php.inc#L22

**Comment by TomasVotruba** [comment]
Tests is passing :+1:

**Comment by zingimmick** [comment]
> Tests is passing 👍


This test fixture failed when use `assertStringEqualsFile` but passed when use `assertStringMatchesFormat`

**Comment by zingimmick** [comment]
Test failed at line 120 but passed at line 129
https://github.com/rectorphp/rector-src/blob/a8ccfeb3d5cf251ae0c87f2bd9a461cd4bd84291/packages/Testing/PHPUnit/AbstractRectorTestCase.php#L119-L130

File: rules-tests/CodingStyle/Rector/Encapsed/EncapsedStringsToSprintfRector/Fixture/encapsed_strings_to_sprintf_should_escape_percent.php.inc [hunk 1]
```diff
@@ -19,7 +19,7 @@ final class EncapsedStringsToSprintfShouldEscapePercent
 {
     public function run(string $value)
     {
-        return sprintf('%s%', $value);
+        return sprintf('%s%%', $value);
```
Comment (TomasVotruba) [inline rules-tests/CodingStyle/Rector/Encapsed/EncapsedStringsToSprintfRector/Fixture/encapsed_strings_to_sprintf_should_escape_percent.php.inc:22]: This now removes former test fixture. Could you add a new file so we have them both?

Comment (zingimmick) [reply rules-tests/CodingStyle/Rector/Encapsed/EncapsedStringsToSprintfRector/Fixture/encapsed_strings_to_sprintf_should_escape_percent.php.inc:22]: But former test fixture was originally like this, Ref: rectorphp/rector#5176.

Pattern `%s%` is not valid.

Comment (zingimmick) [reply rules-tests/CodingStyle/Rector/Encapsed/EncapsedStringsToSprintfRector/Fixture/encapsed_strings_to_sprintf_should_escape_percent.php.inc:22]: A new file added.

Comment (TomasVotruba) [reply rules-tests/CodingStyle/Rector/Encapsed/EncapsedStringsToSprintfRector/Fixture/encapsed_strings_to_sprintf_should_escape_percent.php.inc:22]: So the old fixture should not work?
If so, then it's ok to have just one. I thought the old fixture is correct

**Comment by TomasVotruba** [comment]
Thanks!

Could you come up with a fix too?

**Comment by zingimmick** [comment]
> Thanks!
> 
> Could you come up with a fix too?

I'll try to fix it later.

File: rules-tests/CodingStyle/Rector/Encapsed/EncapsedStringsToSprintfRector/Fixture/encapsed_strings_to_sprintf_should_escape_percent.php.inc (hunk 1 above)
Comment (zingimmick) [reply rules-tests/CodingStyle/Rector/Encapsed/EncapsedStringsToSprintfRector/Fixture/encapsed_strings_to_sprintf_should_escape_percent.php.inc:22]: I just removed the new file and updated the old fixture.

**Comment by zingimmick** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
That was fast :) :+1:  thanks

**Pull request merged**